### ISSUE TITLE
LDAP import hangs if pagePause = true

### ISSUE THREAD

**Issue by jen13501** [backstage/backstage#9527]
## Expected Behavior

[Documentation](https://backstage.io/docs/integrations/ldap/org#users) says paging (both pageSize and pagePause) is supported. PageSize should query the LDAP server in specified batches. If pagePause=true, the plugin should process each batch of entries before requesting the next batch.

## Actual Behavior

pageSize works as expected. pagePause=true causes the processor to get the first batch of entries and then hang. The entries are never processed and the next batch of entries is not returned. 

The LDAP client is missing a response listener for the `page` event [here](https://github.com/backstage/backstage/blob/v0.67.0/plugins/catalog-backend-module-ldap/src/ldap/client.ts#L145) so that when pagePause=true, the `page` event callback is never called. Expected callback behavior described [here](http://ldapjs.org/client.html#paging)

## Steps to Reproduce

1. Add an LDAP integration to app-config.yaml as specified in the [getting started docs](https://backstage.io/docs/integrations/ldap/org)
2. Include `options.paged.pageSize: 10` and `options.paged.pagePause: true` under groups or users (or both!)
3. Observe that LDAP entries are not processed

## Context

Our LDAP data is too large to import all at once, so we must use paging. It'd be nice to see entries as they're pulled from the LDAP server. Even if pagePause support is not added, docs should be updated to reflect that pagePause is not actually supported.

## Your Environment

- Browser Information: 
n/a

- Output of `yarn backstage-cli info`: 

```
OS:   Darwin 19.6.0 - darwin/x64
node: v14.18.3
yarn: 1.22.1
cli:  0.13.1 (local)

Dependencies:
  @backstage/core-plugin-api 0.4.1
✨  Done in 2.01s.

```

**Comment by skfarhat** [comment]
Seconding this issue. I discovered the same behaviour in our LDAP plugin yesterday.

**Comment by freben** [comment]
Ah, thanks for reporting this so thoroughly! Nice.

Since we do streaming ingestion and process each item one by one (per default), the `pagePause` option probably makes little sense to use in practice from Backstage's point of view - is my understanding correct there? Or is there some other reason that one might want to use that option, e.g. for the benefit of the LDAP server or so?

In any case - setting the option should of course not _break_ ingestion! We would love a contribution from either of you to address that problem, if you can. Would it be sufficient to just add a `.on('page', () => {})` that does absolutely nothing?

**Comment by jen13501** [comment]
I'm happy to add the dummy `page` listener to un-break pagePause=true.

It's not really streaming from the LDAP server. It's storing all of the entries in an array before calling `emit` for each (it looks like that's when the catalog is actually told to ingest the data?). I ran with about 29k users this morning and didn't have any issues, but I assume for larger organizations an OOM is likely. I think you'd just want to call the emit function as each entity is received instead of after they've all been received, but I can take a closer look later this week.

**Comment by freben** [comment]
You are right, in the sense that it holds on to all of the output Entity objects. Crucially though, it doesn't hold on to all of the _raw LDAP entries_ during the ingestion cycle; that did indeed lead to memory pressure problems, which was what led to https://github.com/backstage/backstage/pull/7180 . The entities are rather small and simple in comparison. I'll also note here, that the [`LdapOrgEntityProvider`](https://backstage.io/docs/integrations/ldap/org#using-a-provider-instead-of-a-processor) is the much faster, smarter alternative to `LdapOrgReaderProcessor`. It performs lightning fast (in comparison) batch inserts/replacements of the entire set of entities at once, and cleans up after itself. See [this article](https://backstage.io/docs/features/software-catalog/external-integrations) for a description of the differences.

**Comment by freben** [comment]
I believe this was fixed by #9771
